torchvision/prototype/transforms/color.py

```diff
@@ -0,0 +1,116 @@
+import collections.abc
+import functools
+from typing import Any, Dict, Union, Tuple, Optional, Sequence, Callable
+
+import PIL.Image
+import torch
+from torch import nn
+from torchvision.prototype import features
+from torchvision.prototype.transforms import Compose
+from torchvision.prototype.transforms import Transform, functional as F
+
+from ._utils import is_simple_tensor
+
+
+class ColorJitter(Transform):
+    def __init__(
+        self,
+        brightness: Optional[Union[float, Sequence[float]]] = None,
+        contrast: Optional[Union[float, Sequence[float]]] = None,
+        saturation: Optional[Union[float, Sequence[float]]] = None,
+        hue: Optional[Union[float, Sequence[float]]] = None,
+    ) -> None:
+        super().__init__()
+        self.brightness = self._check_input(brightness, "brightness")
+        self.contrast = self._check_input(contrast, "contrast")
+        self.saturation = self._check_input(saturation, "saturation")
+        self.hue = self._check_input(hue, "hue", center=0, bound=(-0.5, 0.5), clip_first_on_zero=False)
+
+    def _check_input(
+        self,
+        value: Optional[Union[float, Sequence[float]]],
+        name: str,
+        center: float = 1.0,
+        bound: Tuple[float, float] = (0, float("inf")),
+        clip_first_on_zero: bool = True,
+    ) -> Optional[Tuple[float, float]]:
+        if value is None:
+            return None
+
+        if isinstance(value, float):
+            if value < 0:
+                raise ValueError(f"If {name} is a single number, it must be non negative.")
+            value = [center - value, center + value]
+            if clip_first_on_zero:
+                value[0] = max(value[0], 0.0)
+        elif isinstance(value, collections.abc.Sequence) and len(value) == 2:
+            if not bound[0] <= value[0] <= value[1] <= bound[1]:
+                raise ValueError(f"{name} values should be between {bound}")
+        else:
+            raise TypeError(f"{name} should be a single number or a sequence with length 2.")
+
+        return float(value[0]), float(value[1])
```

Comment:
You are missing:
https://github.com/pytorch/vision/blob/39772ece7c87ab38b9c2b9df8e7c85e967a739de/torchvision/transforms/transforms.py#L1183-L1186

I understand that you choose to configure `value = None` on the constructor for the same thing, but what you have here is not BC for the case where (0, 0) or (1, 1) etc.

Reply:
Good catch!

Reply:
To be fair, the behavior was not BC breaking. The difference is that the transformations would have been performed with a no-op parametrization. So if they don't have a shortcut for that, we get overhead, but not different results.